### PR TITLE
Make `math.class` affect the limit configuration

### LINKED ISSUE #3230: Changing the math class does not affect default script attachment
### Description

Using [`math.class`](https://typst.app/docs/reference/math/class/) does not affect default script attachment.

For example, the following example renders the same as if the class was not modified:
```typ
$ class("normal", ->)_a $
$ class("relation", x)_a $
```
The first "a" is under the arrow. The second "a" is on the bottom right of the "x".

Instead, I would expect it to render that way:
The first "a" is on the bottom right of the arrow. The second "a" is under the "x".

Moreover, the documentation for `math.class` should be more precise about what it means for a symbol behave like _\<some class>_.

### Reproduction URL

_No response_

### Operating system

_No response_

### Typst version

- [X] I am using the latest version of Typst

### PR DESCRIPTION
**Pull request by MDLC01**
Fixes #3230.

This can always be overridden with [`math.scripts`](https://typst.app/docs/reference/math/attach#functions-scripts) and [`math.limits`](https://typst.app/docs/reference/math/attach#functions-limits).

As suggested in #3230, this PR also improves the documentation for `math.class`.

### CODE REVIEW

**Comment by EpicEricEE** [comment]
Should the special handling for integral symbols with "large" class also be added here?
Otherwise `class("large", integral)_a^b` would probably not give the intended result. (It doesn't really make sense to write, but still)

**Comment by MDLC01** [comment]
I don't think so.

First, this would be complicated to implement properly (the integral symbol might be wrapped in something else) for a situation that would almost never happen.

Second, if I had to give an interpretation to `class("large", integral)`, it would be "make it behave like any non-special large thing".

File: crates/typst/src/math/class.rs [hunk 1]
```diff
@@ -2,12 +2,15 @@ use unicode_math_class::MathClass;
...
 use crate::foundations::{elem, Content, Packed};
-use crate::math::{LayoutMath, MathContext};
+use crate::math::{LayoutMath, Limits, MathContext};
 
 /// Forced use of a certain math class.
 ///
 /// This is useful to treat certain symbols as if they were of a different
-/// class, e.g. to make a symbol behave like a relation.
+/// class, e.g. to make a symbol behave like a relation. The class of a symbol
+/// defines spacing around the symbol, and how scripts are attached to it by
```
Comment (sitandr) [inline crates/typst/src/math/class.rs:11]: If I understand it right, it isn't only the spacing. The symbol also may be moved vertically, and I think there should be more consequences. I think it's important this is not just the spacing.

See this example, where `b` letter moves down when set as `opening` class:

https://sitandr.github.io/typst-examples-book/book/basics/math/classes.html

So I have no idea how to write it in a good way. I haven't look thoroughly in the source, though.

Comment (MDLC01) [reply crates/typst/src/math/class.rs:11]: What about "The class of a symbol defines the way it is laid out, including spacing around it, and how its scripts are attached to by default."?

Comment (sitandr) [reply crates/typst/src/math/class.rs:11]: Sounds good!

**Comment by laurmaedje** [comment]
Thank you!

**Pull request merged**